kong/conf_loader.lua

```diff
@@ -230,7 +224,7 @@ local function check_and_infer(conf)
     end
   end
 
-  if conf.ssl then
+  if table.concat(conf.proxy_listen, ","):find("ssl") then
```

Comment:
This approach still has the issue described yesterday wrt. specifying a domain name that contains "ssl".

Alternatively, to avoid duplicated logic and reduce cognitive load, we could move this validation logic down below, once the listeners are already parsed. After all, there is no need for this validation logic to be part of the inference function it currently belongs to...

Reply:
There should be unit tests for this - it is self-contained and easy to test.

Reply:
the reordering would mean quite a reshuffle of code. So I didn't, this is the lesser of the evils imo. Updated the pattern to prevent false positives (and added tests), but otherwise left as is.

Reply:
Yeah, I can understand that. I wouldn't be too hard to do but that's fine.